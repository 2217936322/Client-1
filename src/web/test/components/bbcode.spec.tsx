import React from 'react';
import { shallow, mount } from 'enzyme';
import BBCode from '@web/components/messageTypes/bbcode/__all__';

describe('bbcode render', () => {
  test.each([
    [':cat:'],
    ['[img]http://example.com[/img]'],
    ['mix text image[img]http://example.com[/img]'],
    ['some sample text'],
    ['[asd]unrecognized tag should be ignore[/asd]'],
  ])('render "%s"', (originText) => {
    const wrapper = mount(<BBCode plainText={originText} />);
    expect(wrapper.toJson()).toMatchSnapshot();
  });
});
