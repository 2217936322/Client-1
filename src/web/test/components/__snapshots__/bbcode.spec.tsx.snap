// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bbcode render render ":cat:" 1`] = `
<Memo()
  plainText=":cat:"
>
  <Memo()
    key="0"
  >
    <pre>
      üê±
    </pre>
  </Memo()>
</Memo()>
`;

exports[`bbcode render render "[img]http://example.com[/img]" 1`] = `
<Memo()
  plainText="[img]http://example.com[/img]"
>
  <ImageTag
    key="0"
    node={
      TagNode {
        "attrs": Object {},
        "content": Array [
          "http://example.com",
        ],
        "tag": "img",
      }
    }
  >
    <mock-image
      role="chatimage"
      src="http://example.com"
    />
  </ImageTag>
</Memo()>
`;

exports[`bbcode render render "mix text image[img]http://example.com[/img]" 1`] = `
<Memo()
  plainText="mix text image[img]http://example.com[/img]"
>
  <Memo()
    key="0"
  >
    <pre>
      mix text image
    </pre>
  </Memo()>
  <ImageTag
    key="1"
    node={
      TagNode {
        "attrs": Object {},
        "content": Array [
          "http://example.com",
        ],
        "tag": "img",
      }
    }
  >
    <mock-image
      role="chatimage"
      src="http://example.com"
    />
  </ImageTag>
</Memo()>
`;
